modules/scaleform.lua

```diff
@@ -0,0 +1,224 @@
+        end
+    end
+end
+
+---@param type string
+---@return boolean | int | string
+---@description Awaits the return value, and converts it to a usable data type
+local function retrieveReturnValue(type)
+    local result = EndScaleformMovieMethodReturnValue() -- End the scaleform movie method with a return value
+
+    lib.waitFor(function()
+        if IsScaleformMovieMethodReturnValueReady(result) then
+            return true
+        end
+    end, "Failed to retrieve return value", 1000)
+
+    if type == "int" then -- If the type is an integer
+        return GetScaleformMovieMethodReturnValueInt(result) -- Get the return value as an integer
+    elseif type == "bool" then
+        return GetScaleformMovieMethodReturnValueBool(result) -- Get the return value as a boolean
+    else -- If the type is not an integer
+        return GetScaleformMovieMethodReturnValueString(result) -- Get the return value as a string
+    end
+end
+
+---@param name string
+---@param args? table
+---@param returnValue? string
+---@return any
+---@description Call a scaleform function, with optional args or return value.
+function Scaleform:method(name, args, returnValue)
+    if not self.handle then -- If the scaleform handle is nil
+        return error('Scaleform handle is nil') -- Error the scaleform handle is nil
+    end
+
+    if args and type(args) ~= 'table' then -- If the type of args is not a table
+        return error('Args must be a table') -- Error args must be a table
+    end
+
+    BeginScaleformMovieMethod(self.handle, name) -- Begin the scaleform movie method
+
+    -- Converts the arguments into data types usable by scaleform
+    if args then
+        convertArgs(args)
+    end
+
+    -- When wanting a return value, the scaleform has to be ended with a different native
+    if returnValue then
+        retrieveReturnValue(returnValue)
+    else
+        EndScaleformMovieMethod() -- End the scaleform movie method
+    end
+end
+
+---@param isFullscreen boolean
+---@return nil
+---@description Set the scaleform to render in full screen
+function Scaleform:setFullScreen(isFullscreen)
+    self.fullScreen = isFullscreen
+end
+
+---@param x number
+---@param y number
+---@param width number
+---@param height number
+---@return nil
+---@description Set the properties of the scaleform (Requires SetFullScreen to be false)
+function Scaleform:setProperties(x, y, width, height)
+    if self.fullScreen then -- If the full screen is true
+        return error('Cannot set properties when full screen is enabled') -- Error cannot set properties when full screen is enabled
+    end
+    self.x = x
+    self.y = y
+    self.width = width
+    self.height = height
+end
+
+---@param name string
+---@param model string|number
+---@return nil
+---@description Create a render target for the scaleform - optional , only if you want to render the scaleform in 3D
+function Scaleform:renderTarget(name, model)
```

Comment:
Can be renamed to something like `setRenderTarget` to have a verb in the beginning.